docs/en/upgradable-smart-contracts-openzeppelin-sdk.md

```diff
@@ -0,0 +1,220 @@
+---
+id: updradable-smart-contracts-with-openzeppelin-sdk
+title: Upgradable Smart Contracts with OpenZeppelin SDK
+sidebar_label: Upgradable Smart Contracts with OpenZeppelin SDK
+---
+
+## Purpose
+
+This guide will walk you through the process of deploying an [upgradable smart contract](https://docs.openzeppelin.com/sdk/2.5/writing-contracts) using the [OpenZeppelin SDK](https://docs.openzeppelin.com/sdk/).
+
+## Prerequisites
+
+- Node.js 10 or higher (recommended v10.15.3). You can download the installer from [this page](https://nodejs.org/en/download/)
+
+- Yarn. Follow [these instructions](https://yarnpkg.com/lang/en/docs/install/) to install Yarn on your computer.
+
+## Setting Things Up
+
+The first step is to install the `OpenZeppelin SDK`. In a terminal window, move into the directory where you want to save the new project, and enter the following commands:
+
+```bash
+mkdir upgradable-contracts && cd upgradable-contracts
+
+# Setting up the packages
+yarn add -D @openzeppelin/cli@2.5.3 js-sha256 loom-js@dev loom-truffle-provider@dev
+
+# Intialize OpenZeppelin SDK
+yarn openzeppelin init upgradable-contracts 1.0.0
+```
+
+> Note: We're using the developer versions of `loom-js` and `loom-truffle-provider`.
+
+The above command creates a bunch of directories and files. For the scope of this tutorial we'll be using:
+
+* The `networks.js`file which stores the deployment configuration. If you're familiar with Truffle, this file is similar to the `truffle-config.js` file.
+* The `contracts` directory where we'll place our smart contracts.
+
+## Configuring LoomTruffleProvider
+
+Now, let's configure OpenZeppelin to use `loom-truffle-provider` and deploy to Loom Testnet (`extdev`). Replace the content of the `networks.js` file with the following snippet:
+
+```js
+const LoomTruffleProvider = require('loom-truffle-provider')
+const { sha256 } = require('js-sha256')
+const { CryptoUtils } = require('loom-js')
+function getLoomProviderWithMnemonic (seed, chainId, writeUrl, readUrl) {
+ const privateKeyUint8ArrayFromSeed = CryptoUtils.generatePrivateKeyFromSeed(new Uint8Array(sha256.array(seed)))
+ const privateKeyB64 = CryptoUtils.Uint8ArrayToB64(privateKeyUint8ArrayFromSeed)
+ return new LoomTruffleProvider(chainId, writeUrl, readUrl, privateKeyB64)
+}
+module.exports = {
+ networks: {
+ extdev: {
+ provider: function() {
+ const chainId = 'extdev-plasma-us1'
+ const writeUrl = 'http://extdev-plasma-us1.dappchains.com:80/rpc'
+ const readUrl = 'http://extdev-plasma-us1.dappchains.com:80/eth'
+ const seed = 'gravity top burden flip student usage spell purchase hundred improve check genre'
+ const loomTruffleProvider = getLoomProviderWithMnemonic(seed, chainId, writeUrl, readUrl)
+ // Quick implementation for web3_clientVersion
+ loomTruffleProvider.getProviderEngine().addCustomMethod('web3_clientVersion', () => {
+ return 'loom-js'
+ })
```

Comment:
This can be written more concisely:
```js
loomTruffleProvider.getProviderEngine().addCustomMethod('web3_clientVersion', () => 'loom-js')
```

Reply:
This should be wiped out actually, in loom-js there's a PR for return the `web3_clientVersion`

Reply:
@enlight @eduardonunesp It's updated.